Rework parameter count inspection (fixes #2115)

dlsniper opened the pull request:
@ignatov not sure if this is what you had in mind but let me know if I need to rework anything else

dlsniper committed:
Rework parameter count inspection (fixes #2115, #2551, #1577)

ignatov commented on .idea/codeStyleSettings.xml:
```diff
@@ -225,4 +225,4 @@
   </component>
-</project>
\ No newline at end of file
+</project>
```
Hmm, it's definitely from another change list.

ignatov commented on gradle.properties:
```diff
@@ -26,4 +26,4 @@ sources=true
 publishChannel=
-org.gradle.jvmargs=-Xms256m -Xmx1024m
\ No newline at end of file
+org.gradle.jvmargs=-Xms256m -Xmx1024m
```
And this one.

dlsniper commented:
Removed unrelated misc file changes

dlsniper commented:
Once this gets merged I plan to work on fixing: https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/1968 as currently this PR doesn't cover it. However I want to keep the PR smaller than the one with assignment count mismatch so I'd rather handle one issue at a time as it appears this time around. Thank you

dlsniper commented:
Updated to be mergeable again (I'll update it as soon as it changes state if again).

dlsniper commented:
Updated

dlsniper commented:
Updated

dlsniper commented:
Updated this as well.

ignatov commented on testData/highlighting/funcCall.go:
```diff
@@ -143,8 +143,8 @@ func bar_m() (int, int) {
 
-func <warning>main127</warning>() {
-	foo_m<error>(bar_m())</error>
+func <warning descr="Unused function 'main127'">main127</warning>() {
```
Please don't add such markup without need, it's very hard to review.

dlsniper replied:
I'll try and remove those, I've understood meanwhile why they appear in the tests since I've wrote this :)

ignatov commented on tests/com/goide/inspections/GoHighlightingTest.java:
```diff
@@ -135,6 +135,7 @@ protected String getBasePath() {
   public void testEmbeddedInterfacePointer()  { doTest(); }
   public void testPlaceholderCount()          { doTest(); }
   public void testTypeConversion()            { doTest(); }
+  public void testGh2115()                    { doTest(); }
```
Better to create a test with a meaningful name, not a issue number.

dlsniper replied:
I'll update it, yes.

dlsniper commented:
Done, I've fixed the previous comments, added the extra test cases from the other two related issues and made the suggestion @zolotov had regarding the position of where the error is reported, it's now on the function name not the the whole arguments list (which makes it more visible when there are no arguments).

dlsniper commented:
The only thing that's different is the error message vs the Go compiler in some cases but I think it's more readable this way (I can use the compiler messages if needed).
